### PR TITLE
Feature/io 585 flavors

### PR DESCRIPTION
**Pull request by ekaterina-demeshko-idf**

### CODE REVIEW

File: packages/presentation/lib/screen/home/home_screen.dart [hunk 1]
```diff
@@ -56,7 +56,7 @@ class _HomeScreenState extends BlocScreenState<HomeScreen, HomeBloc>
           backgroundColor: PrimaryColors.primaryBackgroundColor,
           elevation: 0,
           title: Text(
-            AppLocalizations.of(context)!.title,
+            AppLocalizations.of(context)!.title, //AppConfig.of(context).appTitle
```
Comment (nikolay-ahramovich-idf) [inline packages/presentation/lib/screen/home/home_screen.dart]: Remove comments

File: packages/presentation/lib/screen/home/widgets/movie_list.dart [hunk 2]
```diff
@@ -42,92 +43,102 @@ class _MovieListWidgetState extends BlocScreenState<MovieListWidget, HomeBloc> {
...
+                                          SvgPicture.asset(ImagesPath.starHalf),
+                                      empty: SvgPicture.asset(
+                                          ImagesPath.starEmpty),
+                                    ),
+                                    itemPadding: const EdgeInsets.only(
+                                        right: Dimens.size2),
+                                    onRatingUpdate: (rating) {},
+                                  ),
+                                  const SizedBox(
+                                    height: 8,
```
Comment (nikolay-ahramovich-idf) [inline packages/presentation/lib/screen/home/widgets/movie_list.dart]: Remove comma, and check other SizedBox with one argument

File: packages/presentation/pubspec.yaml [hunk 3]
```diff
@@ -17,6 +17,8 @@ dependencies:
   intl: ^0.17.0
   domain:
     path: ../domain
+#  root:
+#    path: ../../../movie
```
Comment (nikolay-ahramovich-idf) [inline packages/presentation/pubspec.yaml]: Remove comments

**Review by nikolay-ahramovich-idf — APPROVED** [review]

File: lib/di/app_injector.dart [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
+import 'package:movie/flavors/config_presentation.dart';
 import 'package:presentation/di/presentation_injector.dart';
 
-void initAppInjector() {
-  initDataInjector();
+void initAppInjector(
+  ConfigData configData,
+  ConfigPresentation configPresentation,
+) {
+  initDataInjector(configData.apiKey, configData.baseUrl);
```
Comment (sergei-mikhailovskii-idf) [inline lib/di/app_injector.dart]: pass `configData` to `initDataInjector`

File: lib/flavors/config_data.dart [hunk 5]
```diff
@@ -0,0 +1,13 @@
+class ConfigData {
```
Comment (sergei-mikhailovskii-idf) [inline lib/flavors/config_data.dart]: this class should be in the data, not in the root

File: lib/flavors/config_presentation.dart [hunk 6]
```diff
@@ -0,0 +1,8 @@
+class ConfigPresentation {
```
Comment (sergei-mikhailovskii-idf) [inline lib/flavors/config_presentation.dart]: -//-

File: lib/main.dart [hunk 7]
```diff
@@ -1,9 +1,29 @@
...
 import 'di/app_injector.dart';
 
-void main() {
+void mainCommon({Environment env = Environment.production}) async {
   WidgetsFlutterBinding.ensureInitialized();
-  initAppInjector();
-  runApp(const MyApp());
-}
\ No newline at end of file
+  final currentEnv = env == Environment.production ? 'prod' : 'sandbox';
```
Comment (sergei-mikhailovskii-idf) [inline lib/main.dart]: use `describeEnum` method

File: lib/main.dart [hunk 8]
```diff
@@ -1,9 +1,29 @@
...
+  final ConfigData configData = ConfigData.fromJson(await readJson(currentEnv));
```
Comment (sergei-mikhailovskii-idf) [inline lib/main.dart]: move `await readJson(currentEnv)` to separate variable as it is used in several places

File: packages/presentation/lib/screen/home/widgets/movie_list.dart (hunk 2 above)
Comment (sergei-mikhailovskii-idf) [reply packages/presentation/lib/screen/home/widgets/movie_list.dart]: still actual

File: packages/presentation/lib/screen/home/widgets/movie_list.dart [hunk 9]
```diff
@@ -42,92 +43,102 @@ class _MovieListWidgetState extends BlocScreenState<MovieListWidget, HomeBloc> {
...
+                                    maxLines: 1,
+                                    overflow: TextOverflow.ellipsis,
+                                  ),
+                                  const SizedBox(
+                                    height: Dimens.size4,
+                                  ),
+                                  Text(
+                                    '${currentMovie.genre} · ${currentMovie.runtime} | ${currentMovie.certification}',
+                                    style: AppTextStyles.descriptionStyle(
+                                        AppTextStyles.fontSize_14),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/home/widgets/movie_list.dart]: comma or in one line

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: lib/flavors/config_data.dart (hunk 5 above)
Comment (ekaterina-demeshko-idf) [reply lib/flavors/config_data.dart]: how then I will pass it in initDataInjector? the  main lib layer does not has connection with data layer.

File: lib/di/app_injector.dart (hunk 4 above)
Comment (ekaterina-demeshko-idf) [reply lib/di/app_injector.dart]: initDataInjector don't know about ConfigData model

File: lib/main.dart (hunk 7 above)
Comment (ekaterina-demeshko-idf) [reply lib/main.dart]: or maybe  ? Environment.production.name
      : Environment.sandbox.name;

Comment (sergei-mikhailovskii-idf) [reply lib/main.dart]: check [this](https://api.flutter.dev/flutter/foundation/describeEnum.html) and let's discuss after if it is needed

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Review by andrei-lastochkin-idf — APPROVED** [review]

**Pull request merged**